src/main/java/org/springframework/data/redis/connection/ClusterTopology.java

```diff
@@ -79,13 +81,9 @@ public Set<RedisClusterNode> getActiveNodes() {
 	 */
 	public Set<RedisClusterNode> getActiveMasterNodes() {
 
-		Set<RedisClusterNode> activeMasterNodes = new LinkedHashSet<>(nodes.size());
-		for (RedisClusterNode node : nodes) {
-			if (node.isMaster() && node.isConnected() && !node.isMarkedAsFail()) {
-				activeMasterNodes.add(node);
-			}
-		}
-		return activeMasterNodes;
+		return getNodes().stream()
```

Comment:
Stream usage introduces a huge performance penalty due to its excessive object allocation. We avoid stream usage to not accidentally introduce a performance regression unless code paths are guaranteed to run once or during the initialization phase.

Reply:
Agreed

Reply:
Reverted, completely.